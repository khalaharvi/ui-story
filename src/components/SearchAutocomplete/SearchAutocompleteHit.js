import React from 'react';
import PropTypes from 'prop-types';

import { cx } from './SearchAutocomplete';

const SearchAutocompleteHit = ({ hit, hitTitleKey, hitImageKey }) => {
  const titleKey = typeof hitTitleKey === 'string' ? hitTitleKey : hitTitleKey.find(key => !!hit[key]);
  let hitImg = null;

  if (hitImageKey) {
    const imgKey = (typeof hitImageKey === 'string') ? hitImageKey : hitImageKey.find(key => !!hit[key]);

    hitImg = imgKey && hit[imgKey] ?
      <div className={cx('hit-image')}><img src={hit[imgKey]} alt={hit[titleKey]} /></div> :
      <div className={cx('hit-image')}><img src="http://via.placeholder.com/40" alt="placeholder" /></div>;
  }

  return (
    <div className={cx('hit')}>{hitImg}<span className={cx('hit-title')}>{hit[titleKey]}</span></div>
  );
};

SearchAutocompleteHit.propTypes = {
  hit: PropTypes.object.isRequired, // eslint-disable-line react/forbid-prop-types
  hitTitleKey: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.arrayOf(PropTypes.string)
  ]).isRequired,
  hitImageKey: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.arrayOf(PropTypes.string)
  ])
};

export default SearchAutocompleteHit;
