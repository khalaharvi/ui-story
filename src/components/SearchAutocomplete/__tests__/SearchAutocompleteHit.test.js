import React from 'react';
import { mount, shallow } from 'enzyme';

import SearchAutocompleteHit from '../SearchAutocompleteHit';

const hit = {
  title: 'Hit Title',
  image: 'https://placehold.it/50x50'
};
const hitTitleKey = 'title';
const hitImageKey = 'image';

describe('<SearchAutocompleteHit />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<SearchAutocompleteHit hit={hit} hitTitleKey={hitTitleKey} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly with hitTitleKey array prop', () => {
    const wrapper = shallow(<SearchAutocompleteHit hit={hit} hitTitleKey={['title', 'name']} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly with hitImageKey prop', () => {
    const wrapper = shallow(<SearchAutocompleteHit
      hit={hit}
      hitTitleKey={hitTitleKey}
      hitImageKey={hitImageKey}
    />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly with hitImageKey array prop', () => {
    const wrapper = shallow(<SearchAutocompleteHit
      hit={hit}
      hitTitleKey={hitTitleKey}
      hitImageKey={['image', 'primary_image']}
    />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should have an <img /> when hitImageKey prop is passed', () => {
    const wrapper = mount(<SearchAutocompleteHit
      hit={hit}
      hitTitleKey={hitTitleKey}
      hitImageKey={hitImageKey}
    />);

    const img = wrapper.find('img');

    expect(img).toHaveLength(1);

    wrapper.unmount();
  });

  it('should have an <img /> when hitImageKey array prop is passed', () => {
    const wrapper = mount(<SearchAutocompleteHit
      hit={hit}
      hitTitleKey={hitTitleKey}
      hitImageKey={['image', 'primary_image']}
    />);

    const img = wrapper.find('img');

    expect(img).toHaveLength(1);

    wrapper.unmount();
  });

  it("shouldn't have an <img /> when 'hitImageKey' key doesn't exist", () => {
    const wrapper = mount(<SearchAutocompleteHit
      hit={hit}
      hitTitleKey={hitTitleKey}
      hitImageKey="primary_image"
    />);

    const img = wrapper.find('img');

    expect(img).toHaveLength(0);

    wrapper.unmount();
  });
});
