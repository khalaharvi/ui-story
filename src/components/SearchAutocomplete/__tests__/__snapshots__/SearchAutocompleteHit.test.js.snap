// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchAutocompleteHit /> renders correctly 1`] = `
<div
  className="dh-search-autocomplete-hit"
>
  <span
    className="dh-search-autocomplete-hit-title"
  >
    Hit Title
  </span>
</div>
`;

exports[`<SearchAutocompleteHit /> renders correctly with hitImageKey array prop 1`] = `
<div
  className="dh-search-autocomplete-hit"
>
  <div
    className="dh-search-autocomplete-hit-image"
  >
    <img
      alt="Hit Title"
      src="https://placehold.it/50x50"
    />
  </div>
  <span
    className="dh-search-autocomplete-hit-title"
  >
    Hit Title
  </span>
</div>
`;

exports[`<SearchAutocompleteHit /> renders correctly with hitImageKey prop 1`] = `
<div
  className="dh-search-autocomplete-hit"
>
  <div
    className="dh-search-autocomplete-hit-image"
  >
    <img
      alt="Hit Title"
      src="https://placehold.it/50x50"
    />
  </div>
  <span
    className="dh-search-autocomplete-hit-title"
  >
    Hit Title
  </span>
</div>
`;

exports[`<SearchAutocompleteHit /> renders correctly with hitTitleKey array prop 1`] = `
<div
  className="dh-search-autocomplete-hit"
>
  <span
    className="dh-search-autocomplete-hit-title"
  >
    Hit Title
  </span>
</div>
`;
